### DISCUSSION TITLE
ESLint no-use-before-define when defining models

### DISCUSSION THREAD

**Discussion by MarioRomanDono** [sequelize/sequelize#16080] [Help & Questions] [answered]
I'm trying to use Sequelize with Typescript as described [in your docs](https://sequelize.org/docs/v6/other-topics/typescript/), but I'm getting an ESLint no-use-before-define error when defining my models. Here is an example:

```
import {
    Model, Sequelize, DataTypes, InferAttributes, InferCreationAttributes
} from 'sequelize';

export default class SequelizeTax extends Model<InferAttributes<SequelizeTax>, InferCreationAttributes<SequelizeTax>> {
    declare id: number;

    declare name: string;

    static initialize(sequelize: Sequelize) {
        this.init({
            id: {
                type: DataTypes.INTEGER.UNSIGNED,
                autoIncrement: true,
                primaryKey: true,
                allowNull: false
            },
            name: {
                type: DataTypes.STRING,
                allowNull: false
            }
        }, {
            sequelize,
            name: {
                singular: 'Tax',
                plural: 'Taxes'
            }
        });
    }
}
```
![](https://i.imgur.com/jPry0Ko.png)

I want to know if I'm doing anything wrong or if I could safely disable the ESLint rule in that line. Thanks!

**Comment by ephys** [comment, marked as answer]
You can try using typescript's version: https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-use-before-define.md (maybe with "ignoreTypeReferences")

If that doesn't work, it's an issue with the lint rule as this is the only way to achieve what we need

**Reply by MarioRomanDono** [reply]
Oops, I went to the typescript-eslint docs but it seems I've looked at it wrong 😅 Thank you!
